.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,92 @@
+name: Build, Test and Deploy
+# Controls when the action will run. Triggers the workflow on push
+on:
+  push:
+    branches:
+      - "*"
+# A workflow run is made up of one or more jobs that can run sequentially or parallel
+jobs:
+  # The first job in this workflow is named build-image
+  build:
+    # Type of runner that the job wll run on
+    runs-on: ubuntu-latest
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      - name: Step 1 - Checkout Github Repository
+        uses: actions/checkout@v3
+      # Add parallel builds for multi platform support
+      - name: Step 2 - Set up QEMU
+        uses: docker/setup-qemu-action@v2
+      # The BuildKit Cache
+      - name: Step 3 - Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      # Push to Docker Engine
+      - name: Step 4 - Build Image & Export to Docker Engine
+        uses: docker/build-push-action@v2
+        with:
+          push: false
+          tags: nblocks-ts-client:latest
+          target: test
+          outputs: type=docker,dest=/tmp/nblocks-ts-client.tar
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          platforms: linux/amd64
+      # Upload Docker Image Artifact
+      - name: Step 5 - Upload Artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: nblocks-ts-client
+          path: /tmp/nblocks-ts-client.tar
+  test:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      # Checkout Github Repository
+      - name: Step 1 - Checkout Github Repository
+        uses: actions/checkout@v3
+      # Add parallel builds for multi platform support
+      - name: Step 2 - Set up QEMU
```

Comment:
Is Qemu and Buildx (Step 2 & Step 3) really necessary even though we just pull an existing image and run it?

Reply:
I don't think we need it. When we talked about it I understood that I should leave it for later, testing the build on other arch...

Reply:
Would you like me to remove them for this merge?

Reply:
Yes, if they are not needed it’s clearer if they’re removed or at least commented out (with a comment describing why)